graph/src/ipfs_client.rs

```diff
@@ -1,14 +1,74 @@
 use crate::prelude::CheapClone;
+use anyhow::anyhow;
 use anyhow::Error;
 use bytes::Bytes;
+use cid::Cid;
 use futures03::Stream;
 use http::header::CONTENT_LENGTH;
 use http::Uri;
 use reqwest::multipart;
 use serde::Deserialize;
+use std::fmt::Display;
 use std::time::Duration;
 use std::{str::FromStr, sync::Arc};
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct CidFile {
+    pub cid: Cid,
+    pub path: Option<String>,
```

Comment:
Lets add a comment above this field clarifying that it never contains the leading slash.